tools/integration_tests/list_large_dir/list_dir_with_twelve_thousand_files_test.go

```diff
@@ -56,36 +144,41 @@ func createHundredExplicitDir(dirPath string, t *testing.T) {
 	}
 }
 
+////////////////////////////////////////////////////////////////////////
+// Tests
+////////////////////////////////////////////////////////////////////////
+
 // Test with a bucket with twelve thousand files.
 func TestListDirectoryWithTwelveThousandFiles(t *testing.T) {
 	createTwelveThousandFilesAndUploadOnTestBucket(t)
 	testDirPath := path.Join(setup.MntDir(), DirectoryForListLargeFileTests)
 	testDirPathOnBucket := path.Join(setup.TestBucket(), DirectoryForListLargeFileTests)
-
 	dirPath := path.Join(testDirPath, DirectoryWithTwelveThousandFiles)
 
+	// List Directory first time
+	startTime := time.Now()
 	objs, err := os.ReadDir(dirPath)
 	if err != nil {
 		t.Errorf("Error in listing directory.")
 	}
-
-	// number of objs - 12000
-	if len(objs) != NumberOfFilesInDirectoryWithTwelveThousandFiles {
-		t.Errorf("Listed incorrect number of files from directory: %v, expected 12000", len(objs))
-	}
-
-	// Checking if all the object is File type.
-	for i := 0; i < len(objs); i++ {
-		if objs[i].IsDir() {
-			t.Errorf("Listed object is incorrect.")
-		}
-	}
-
-	for i := 0; i < len(objs); i++ {
-		checkIfObjNameIsCorrect(objs[i].Name(), PrefixFileInDirectoryWithTwelveThousandFiles, NumberOfFilesInDirectoryWithTwelveThousandFiles, t)
+	endTime := time.Now()
+	validateDirectoryWithTwelveThousandFiles(objs, t)
+	firstListTime := endTime.Sub(startTime)
+	// Listing the directory a second time should retrieve the response from the kernel cache.
+	startTime = time.Now()
+	objs, err = os.ReadDir(dirPath)
+	if err != nil {
+		t.Errorf("Error in listing directory.")
 	}
-
-	// Clear the bucket after testing.
+	endTime = time.Now()
+	validateDirectoryWithTwelveThousandFiles(objs, t)
+	secondListTime := endTime.Sub(startTime)
+
+	// Fetching data from the kernel for the second list will be faster.
+	assert.Less(t, secondListTime, firstListTime)
+	// The second directory listing should take less than 100 milliseconds since it will be retrieved from the kernel cache.
+	assert.Less(t, secondListTime, 100*time.Millisecond)
```

Comment:
How much difference do we see b/w first Listing and 2nd listing?

Reply:
It is in descriptions. Please check.